repair/repair.cc

```diff
@@ -2344,7 +2349,19 @@ future<> repair::tablet_repair_task_impl::run() {
                 task_impl_ptr->neighbors = std::move(neighbors);
                 auto task = co_await rs._repair_module->make_task(std::move(task_impl_ptr), parent_data);
                 task->start();
-                co_await task->done();
+                try {
+                    co_await task->done();
```

Comment:
Use coroutine::as_future(), it will be nicer

Reply:
God idea.  done in v2

Reply:
Why is coroutine::as_future better than the standard try catch? I think it is better to use standard try catch than home-made helpers.

Reply:
try/catch of exceptions is slower.
as_future keeps the exception_ptr in the future state without throwing it to be grabbed by get_exception and returned as to coroutine future result.

Reply:
Since exceptions are taking a global lock they also affect threads running on other cores.
See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=71744 for example.

Reply:
@nyh / @gleb-cloudius what's our position today about exception throwing?
We went out of our way to eliminate exceptions on the data path, but should we try to prevent exceptions also on the maintenance plane?

Reply:
We still throw exceptions. Even on the data path. If there is an easy way around it or re-throw can be avoided then it should be done unless it is on a very slow path and makes the code much less readable.

> Since exceptions are taking a global lock they also affect threads running on other cores.

Not in seastar. Exception still slow, but they computationally slow, not because of a global lock.